packages/tests/server/react/polymorphism.test.tsx

```diff
@@ -263,85 +319,84 @@ describe('polymorphism', () => {
     });
   });
 
-  describe('sub-typed factory', () => {
-    test('can use a sub-typed factory router with the interfaces from the supertype', async () => {
-      const { trpc } = ctx;
-
-      /**
-       * Can define page components which operate over interfaces generated by a super-typed router,
-       * but also extend types and functionality
-       */
-      function PullRequestsExportPage() {
-        const utils = trpc.useUtils();
-
-        const [currentExport, setCurrentExport] = useState<number | null>(null);
-        const invalidate = useMutation({
-          mutationFn: () => utils.invalidate(),
-        });
-
-        return (
-          <>
-            {/* Some components may still need to be bespoke... */}
-            <SubTypedStartExportButton
-              route={trpc.github.pullRequests.export}
-              utils={utils.github.pullRequests.export}
-              onExportStarted={setCurrentExport}
-            />
-
-            <RefreshExportsListButton
-              mutate={invalidate.mutate}
-              isPending={invalidate.isPending}
-            />
-
-            <RemoveExportButton
-              remove={trpc.github.pullRequests.export.delete}
-              utils={utils.github.pullRequests.export}
-              exportId={currentExport}
-            />
-
-            {/* ... or you can adapt them to support sub-types */}
-            <ExportStatus
-              status={trpc.github.pullRequests.export.status}
-              //                                       ^?
-              currentExport={currentExport}
-              renderAdditionalFields={(data) => {
-                expectTypeOf(data.description).toEqualTypeOf<string>();
-                return `Description: "${data?.description}"`;
-              }}
-            />
-
-            <ExportsList list={trpc.github.pullRequests.export.list} />
-          </>
-        );
-      }
-
-      /**
-       * Test Act & Assertions
-       */
-
-      const $ = render(
-        <ctx.App>
-          <PullRequestsExportPage />
-        </ctx.App>,
-      );
-
-      await userEvent.click($.getByTestId('startExportBtn'));
-
-      await waitFor(() => {
-        expect($.container).toHaveTextContent(
-          'Last Export: `Search for Polymorphism React` (Working)',
-        );
-      });
-
-      await userEvent.click($.getByTestId('refreshBtn'));
-
-      await waitFor(() => {
-        expect($.container).toHaveTextContent(
-          'Last Export: `Search for Polymorphism React` (Ready!)',
-        );
-      });
-    });
-  });
+  // describe('sub-typed factory', () => {
```

Comment:
I... don't think subtyping is going to be possible without essentially rewriting the types deeply to remove all callbacks which reference the input or output types

Woulds love to be proven wrong, but structural typing is known not to function when there's an inner callback and the strucutral-ness ends up going the wrong direction and breaking 😔

<img width="324" alt="image" src="https://github.com/user-attachments/assets/e7651163-4598-4caa-8299-012928dd0613">

Reply:
I'm not sure the exact cause in this case but I've seen this plenty of times and it sucks